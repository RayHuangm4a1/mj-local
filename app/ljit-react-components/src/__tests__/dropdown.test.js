import React from 'react';
import { shallow, mount, } from 'enzyme';
import Dropdown from '../components/dropdown';


function renderDropdownItem() {
	return (
		<ul>
			<li>item 1</li>
			<li>item 2</li>
		</ul>
	);
}

describe('Dropdown', () => {
	it('handle default props', () => {
		const {
			placement,
			trigger,
			isDisabled,
			onVisibleChange,
			isShowingArrow,
			isKeepMenuOpen,
		} = Dropdown.defaultProps;

		expect(placement).toEqual(Dropdown.PlacementEnums.BOTTOM_LEFT);
		expect(trigger).toEqual(['hover',]);
		expect(isDisabled).toEqual(false);
		expect(onVisibleChange).toBeDefined();
		expect(onVisibleChange).toBeInstanceOf(Function);
		expect(isShowingArrow).toEqual(false);
		expect(isKeepMenuOpen).toEqual(false);
	});

	it('should renders correctly', () => {
		const className = 'mock-classname';
		const placement = Dropdown.PlacementEnums.BOTTOM_CENTER;
		const children = <div>dropdown-content</div>;

		const wrapper = shallow(
			<Dropdown
				dropdownContent={renderDropdownItem()}
				className={className}
				placement={placement}
				isShowingArrow={false}
				isKeepMenuOpen={false}
			>
				{children}
			</Dropdown>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const className = 'mock-classname';
		const placement = Dropdown.PlacementEnums.BOTTOM_CENTER;
		const children = <div>children</div>;
		const dropdownContent =  renderDropdownItem();
		const trigger = ['click', 'hover',];
		const isDisabled = false;
		const onVisibleChange = () => {};
		const isShowingArrow = false;
		const isKeepMenuOpen = false;
		const props = {
			className,
			dropdownContent,
			placement,
			trigger,
			isDisabled,
			onVisibleChange,
			isShowingArrow,
			isKeepMenuOpen,
		};

		const wrapper = mount(
			<Dropdown {...props}>
				{children}
			</Dropdown>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().dropdownContent).toBe(dropdownContent);
		expect(wrapper.props().placement).toBe(placement);
		expect(wrapper.props().trigger).toBe(trigger);
		expect(wrapper.props().isDisabled).toBe(isDisabled);
		expect(wrapper.props().children).toBe(children);
		expect(wrapper.props().onVisibleChange).toBe(onVisibleChange);
		expect(wrapper.props().isShowingArrow).toBe(isShowingArrow);
		expect(wrapper.props().isKeepMenuOpen).toBe(isKeepMenuOpen);
	});

	describe('when Dropdown.props.trigger has "click" ', () => {
		it('should call onVisibleChange when children is click', () => {
			const mockOnVisibleChange = jest.fn();
			const wrapper = mount(
				<Dropdown
					dropdownContent={renderDropdownItem()}
					trigger={['click',]}
					onVisibleChange={mockOnVisibleChange}
				>
					<button className="mock-child">children</button>
				</Dropdown>
			);

			wrapper.find('.mock-child').simulate('click');
			expect(mockOnVisibleChange).toHaveBeenCalled();
		});
	});
});

describe('PlacementEnums ', () => {
	it('should contains BOTTOM_LEFT property', () => {
		const typeName = 'bottomLeft';
		const formatType = 'BOTTOM_LEFT';

		expect(Dropdown.PlacementEnums[formatType]).toEqual(typeName);
	});

	it('should contains BOTTOM_CENTER property', () => {
		const typeName = 'bottomCenter';
		const formatType = 'BOTTOM_CENTER';

		expect(Dropdown.PlacementEnums[formatType]).toEqual(typeName);
	});
	it('should contains BOTTOM_RIGHT property', () => {
		const typeName = 'bottomRight';
		const formatType = 'BOTTOM_RIGHT';

		expect(Dropdown.PlacementEnums[formatType]).toEqual(typeName);
	});
	it('should contains TOP_LEFT property', () => {
		const typeName = 'topLeft';
		const formatType = 'TOP_LEFT';

		expect(Dropdown.PlacementEnums[formatType]).toEqual(typeName);
	});
	it('should contains TOP_CENTER property', () => {
		const typeName = 'topCenter';
		const formatType = 'TOP_CENTER';

		expect(Dropdown.PlacementEnums[formatType]).toEqual(typeName);
	});
	it('should contains TOP_RIGHT property', () => {
		const typeName = 'topRight';
		const formatType = 'TOP_RIGHT';

		expect(Dropdown.PlacementEnums[formatType]).toEqual(typeName);
	});
});

describe('TriggerEnums ', () => {
	it('should contains CLICK property', () => {
		const typeName = 'click';
		const formatType = 'CLICK';

		expect(Dropdown.TriggerEnums[formatType]).toEqual(typeName);
	});

	it('should contains HOVER property', () => {
		const typeName = 'hover';
		const formatType = 'HOVER';

		expect(Dropdown.TriggerEnums[formatType]).toEqual(typeName);
	});
});

