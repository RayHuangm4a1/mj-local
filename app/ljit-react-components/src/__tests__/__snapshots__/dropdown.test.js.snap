// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    className="mock-classname"
    dropdownContent={
      <ul>
        <li>
          item 1
        </li>
        <li>
          item 2
        </li>
      </ul>
    }
    isDisabled={false}
    isKeepMenuOpen={false}
    isShowingArrow={false}
    onVisibleChange={[Function]}
    placement="bottomCenter"
    trigger={
      Array [
        "hover",
      ]
    }
  >
    <div>
      dropdown-content
    </div>
  </Dropdown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        dropdown-content
      </div>,
      "disabled": false,
      "mouseEnterDelay": 0.15,
      "mouseLeaveDelay": 0.1,
      "onVisibleChange": [Function],
      "overlay": <ul>
        <li>
          item 1
        </li>
        <li>
          item 2
        </li>
      </ul>,
      "overlayClassName": "ljit-dropdown mock-classname",
      "placement": "bottomCenter",
      "trigger": Array [
        "hover",
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "dropdown-content",
      },
      "ref": [Function],
      "rendered": "dropdown-content",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          dropdown-content
        </div>,
        "disabled": false,
        "mouseEnterDelay": 0.15,
        "mouseLeaveDelay": 0.1,
        "onVisibleChange": [Function],
        "overlay": <ul>
          <li>
            item 1
          </li>
          <li>
            item 2
          </li>
        </ul>,
        "overlayClassName": "ljit-dropdown mock-classname",
        "placement": "bottomCenter",
        "trigger": Array [
          "hover",
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "dropdown-content",
        },
        "ref": [Function],
        "rendered": "dropdown-content",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
